### ISSUE TITLE
Core Services: 3.4 Tracking Issue

### ISSUE THREAD

**Issue by keegancsmith** [sourcegraph/sourcegraph#3467]
### Goals

- All supported code hosts have moved to the new model of syncing repository
  metadata. This allows us to support detecting deletions and renames,
  avoiding expensive code host API requests on a user request path and solves
  many related repository metadata bugs.
- A better model for repositoryPathPattern. Also known as aliases for
  repository names. This allows users to have more succinct names for
  repositories on their Sourcegraph instance. Additionally the use of the
  feature won't break the browser extension.
- Onboard Thorsten.

### Availability

- @keegancsmith is out from 19th of April until the 26th.
- @tsenart is out from the 15th of May until the 22nd.
- @mrnugget is joining on the 23rd of April.

We are aiming for **17d** person days of planned work. This is to account for
responding to customer issues, vacation, helping other teams and onboarding
Thorsten.

### Planned Work

- Integrate remaining code hosts with the new syncing model
  - [x] Gitolite #3464  (@tomas) 
  - [x] AWS CodeCommit #3465 (@mrnugget)
- [x] RepoLookup should load metadata from database #3466 (@mrnugget)
- [x] Support new syncer in sourcegraph.com deployment #3459 (@mrnugget, @keegancsmith)
  - ~~[ ] Add git repos to Sourcegraph.com #3457~~
  - [x] Sourcegraph.com does not auto add soft-deleted repositories. Will only be
    an issue once new syncer is enabled. #2933 
- [x] Remove syncer feature flag and delete old code paths #4000 (@tomas)
- [x] Repo name mappings are not universally supported #462 (@keegancsmith)
  - [x] Impossible to change repositoryPathPattern after adding repositories #2741 
  - [x] GitHub link disappears intermittently in 3.x #2677
- [x] repos: Deprecate enabled/disabled state #3291 (@keegancsmith)

### Stretch Goals

- [ ] enqueue-repo-update should use an ID debt #3345 (@tsenart)
- Visiting a repository for the first time shows "starting to clone" but doesn't start #3447 
- Repo language not being updated https://github.com/sourcegraph/sourcegraph/issues/2586
- Redis
  - Redis AOF database corruption #651
  - Redis AOF file grows unbounded due to frequent container restarts #3300
  - redis: Log spam warning when redis is loading the dataset in memory #2904
  - Redis OOM error prevents sign in or using server #296
- postgresql: (Auto)tune DB pool sizes based on max_connections #3473

### Onboarding @mrnugget

> Note: We can adjust this plan based on Thorsten's preferences.

- **Week 1**: Thorsten should spend first week getting to know people in SF and
understanding the high level picture of Sourcegraph Inc. and
sourcegraph/sourcegraph.

- **Week 2**: Pair with us on whatever we are working on. Likely implementing a code
host #3464 or #3465.

- **Week 3-4**: Continue to learn how things are put together and begin taking on some of the planned work in the iteration or other important issues that can pop up.

**Comment by tsenart** [comment]
It's becoming clear that changing our Phabricator integration to use the new syncer is more work than any other code host we have. Once @keegancsmith is back from holiday we'll reconvene about our plan.

**Comment by keegancsmith** [comment]
Additionally we need to discuss if we can implement syncer dry-run mode this iteration to unblock @beyang: https://github.com/sourcegraph/sourcegraph/issues/2866

**Comment by tsenart** [comment]
We are punting #2573 to the next cycle.

**Comment by keegancsmith** [comment]
I've updated the description. Closing the issue :D
